### PR TITLE
Added travis-encrypt-file script

### PR DESCRIPTION
**Pull request by djspiewak**
Use this instead of `travis-encrypt-file`

### CODE REVIEW

**Review by nicflores — APPROVED** [review]

**Comment by djspiewak** [comment]
[ch7650]

File: bin/travis-encrypt-file [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  echo -en "\033[0;37m$*${NC}" 1>&2
+}
+
+nl() {
+  echo 1>&2
+}
+
+if ! git config travis.slug &>/dev/null; then
+  plain "Please define the "
+  yellow "travis.slug"
```
Comment (dcsobral) [inline bin/travis-encrypt-file:38]: Any reason not to have defined the colors as variables, and then just interpolated? Something like
```bash
cat <<-EOF
        Please define the ${yellow}travis.slug${nc} value using ${yellow}git config${nc}
        (or by running ${yellow}travis encrypt{$nc} on something). ...
EOF
```
Or, alternatively, keep the functions but write it like this?
```bash
cat <<-EOF
        Please define the $(yellow travis.slug) value using $(yellow git config)
        (or by running $(yellow travis encrypt) on something). ...
EOF
```
Speaking of scripts, I'm slowly going through scripts and ensuring they pass [shellcheck](https://github.com/koalaman/shellcheck), are formatted to [shfmt](https://github.com/mvdan/sh) tastes (with a few options), and I plan on adding unit testing with [osht](https://github.com/mvdan/sh) next.

Travis has shellcheck pre-installed, so it might be interesting to add it as a step.

Comment (djspiewak) [reply bin/travis-encrypt-file:38]: > Any reason not to have defined the colors as variables, and then just interpolated? 

I find this way a lot clearer. Or at least I did at first. It's gotten a bit bigger and the newlines make it hard to follow, so I probably might have interpolated had I realized it was going to get this big when I started.

> Travis has shellcheck pre-installed, so it might be interesting to add it as a step.

I'm a little leery of linting the formatting, since we don't do that with anything else right now. I'm fine with linting other things though.

File: bin/travis-encrypt-file [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  plain " value using "
+  yellow "git config"
```
Comment (dcsobral) [inline bin/travis-encrypt-file:40]: `git config --local`, preferably... Maybe run `git config --local --add travis.slug $(git remote get-url upstream | sed -Ee 's/.*:(.*).git/\1/')` to auto-set it?

Comment (djspiewak) [reply bin/travis-encrypt-file:40]: I thought about auto-setting, but I didn't want to go too crazy. This tool needs the `travis` CLI anyway, so it's easy enough to delegate it to that. Also `git-config` defaults to `--local`, though we could add it just to be explicit.
